domain/src/main/java/co/nimblehq/domain/data/error/ValidateError.kt

```diff
@@ -0,0 +1,10 @@
+package co.nimblehq.domain.data.error
+
+sealed class ValidateError(
+    cause: Throwable?
+) : AppError(cause) {
+
+    class InvalidEmailError(cause: Throwable?) : ValidateError(cause)
+
+    class InvalidPasswordError(cause: Throwable?) : ValidateError(cause)
```

Comment:
We can change this to `data class` to be easier to check its type. 

Ref: https://kotlinlang.org/docs/reference/sealed-classes.html